src/transformers/models/cohere/modeling_cohere.py

```diff
@@ -1021,8 +1021,11 @@ def _update_causal_mask(
             causal_mask = causal_mask.clone()  # copy to contiguous memory for in-place edit
             if attention_mask.dim() == 2:
                 mask_length = attention_mask.shape[-1]
-                padding_mask = causal_mask[..., :mask_length].eq(0.0) * attention_mask[:, None, None, :].eq(0.0)
-                causal_mask[..., :mask_length] = causal_mask[..., :mask_length].masked_fill(padding_mask, min_dtype)
+                padding_mask = causal_mask[:, :, :, :mask_length] + attention_mask[:, None, None, :]
```

Comment:
you risk overflows here no? 
If you have torch_min + torch_min -> produced inf

Reply:
+1 - we should try to avoid this. I think it's technically OK because of the masked_fill below

Reply:
Even if you overflow it's ok because the next line you create a boolean tensor: `padding_mask = padding_mask == 0`. So in the end you get the same result as long as the elements in the tensor that need to be zero end up having this value.

This code, which overflows in the context of the function:
```python
padding_mask = causal_mask[:, :, :, :mask_length] + attention_mask[:, None, None, :]
padding_mask *= 2
padding_mask = padding_mask == 0
```

produces the same output as this code, which does not overflow:

```python
padding_mask = causal_mask[:, :, :, :mask_length] + attention_mask[:, None, None, :]
padding_mask = padding_mask == 0
```